lib/src/report/template.dart

```diff
@@ -48,26 +47,30 @@ Future<ReportSection> followsTemplate(PackageContext context) async {
       return issues;
     }
 
-    final status = await context.urlChecker.checkStatus(
-      url,
-      isInternalPackage: options.isInternal,
-    );
-    if (status == UrlStatus.invalid || status == UrlStatus.internal) {
+    final status = await context.urlChecker.checkStatus(url);
+    if (status.isInvalid) {
+      issues.add(
+        Issue(
+          "$name isn't valid.",
+          span: tryGetSpanFromYamlMap(pubspec.originalYaml, key),
+        ),
+      );
+    } else if (status.isInternal && !options.isInternal) {
       issues.add(
         Issue(
           "$name isn't helpful.",
           span: tryGetSpanFromYamlMap(pubspec.originalYaml, key),
         ),
       );
-    } else if (status == UrlStatus.missing) {
+    } else if (!status.exists) {
       issues.add(
         Issue(
           "$name doesn't exist.",
           span: tryGetSpanFromYamlMap(pubspec.originalYaml, key),
           suggestion: 'At the time of the analysis `$url` was unreachable.',
         ),
       );
-    } else if (status == UrlStatus.exists && !url.startsWith('https://')) {
+    } else if (status.exists && !status.isSecure) {
```

Comment:
I think in this case you can assume status.exists
```suggestion
    } else if (!status.isSecure) {
```